rewrite-maven/src/main/java/org/openrewrite/maven/cache/InMemoryCache.java

```diff
@@ -53,7 +52,7 @@ private void fillUnresolvablePoms() {
         new BufferedReader(new InputStreamReader(MavenDownloader.class.getResourceAsStream("/unresolvable.txt"), StandardCharsets.UTF_8))
                 .lines()
                 .filter(line -> !line.isEmpty())
-                .forEach(gav -> pomCache.put(gav, Optional.empty()));
+                .forEach(unresolvablePoms::add);
```

Comment:
I originally intended for `unresolvablePoms` to be an immutable denylist. Basically, the bad artifacts that have never worked that regularly get requested in maven central.

Reply:
Seems like we should instead be keying pom cache on gav+repo and adding `Optional.empty()`.

Reply:
I am a little confused about what you want to change here. 

I am collecting the unresolved group/artifact/versions into a string list now.

See below: